lib/remote_persistent_term/fetcher/s3.ex

```diff
@@ -166,14 +168,53 @@ defmodule RemotePersistentTerm.Fetcher.S3 do
 
   defp find_latest(_), do: {:error, :not_found}
 
+  @impl true
+  def previous_version(state) do
+    Logger.info(
+      bucket: state.bucket,
+      key: state.key,
+      message: "About to fetch previous version of object",
+      version_id: state.version_id
+    )
+
+    versions = if state.versions, do: {:ok, state.versions}, else: list_object_versions(state)
```

Comment:
I'm not 100% sure how s3 replication works, are the versions guaranteed to be the same across replicated buckets? If not, then it is possible we have stored the versions of one region, and now are trying a fallback region with the original versions